src/Features/CSharp/Portable/UseRecursivePatterns/CSharpUseRecursivePatternsCodeRefactoringProvider.AnalyzedNode.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseRecursivePatterns
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class CSharpUseRecursivePatternsCodeRefactoringProvider
+    {
+        [Flags]
+        private enum NodeKind
+        {
+            // Represents a combination of either
+            //      two operands of the &&-operator.
+            //      two patterns against the same expression.
+            Conjunction = 1,
```

Comment:
can you give an actual example of these?  same for elow when it's a non-trivial case.  i.e. you don't need to doc 'var pattern', but i would lik ea doc on NotNullPattern.